fe/fe-core/src/main/java/org/apache/doris/nereids/processor/post/ProjectAggregateExpressionsForCse.java

```diff
@@ -102,8 +103,16 @@ public Plan visitPhysicalHashAggregate(PhysicalHashAggregate<? extends Plan> agg
         }
 
         if (aggregate.child() instanceof PhysicalProject) {
+            List<NamedExpression> newProjections = Lists.newArrayList();
             PhysicalProject<? extends Plan> project = (PhysicalProject<? extends Plan>) aggregate.child();
-            List<NamedExpression> newProjections = Lists.newArrayList(project.getProjects());
+            Set<Slot> newInputSlots = aggOutputReplaced.stream()
+                    .flatMap(expr -> expr.getInputSlots().stream())
+                    .collect(Collectors.toSet());
+            for (NamedExpression expr : project.getProjects()) {
+                if (!(expr instanceof SlotReference) || newInputSlots.contains(expr)) {
```

Comment:
why need `!(expr instanceof SlotReference)`?

Reply:
if it is not slot, it is alias. if it is alias, it must be used by Aggregate in 2 cases:
1. directly used by Agg after rewrite
2. used by inferred common expression.
In both cases, alias should not be pruned.